ui/src/components/app-form/app-form.tsx

```diff
@@ -290,7 +510,150 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
       }
     }
   };
+  const handleError = (error: unknown) => {
+    if (error instanceof AxiosError && error.response) {
+      // Error from Axios response
+      setNotification(
+        `Error: ${error.response.status} - ${error.response.data?.message || 'Unknown error'}`,
+      );
+    } else if (error instanceof Error) {
+      // Generic JavaScript error
+      setNotification(`Error: ${error.message}`);
+    } else {
+      // Unknown error
+      setNotification('Unknown error occurred');
+    }
+  };
+
+  const createOrUpdateAppFromGit = (payload: AppFormInput) => {
```

Comment:
This function as well as the `createOrUpdateApp` function is not actually being used, when there are no server types/profiles available (which is generally the case when developing locally). We need to either remove this (as its currently mostly redundant code) or refactor across the board to use these 2 functions.

Reply:
Removed.